### PR TITLE
Swagger Changes For Alerts and Alerts Summary as extension resources

### PR DESCRIPTION
**Pull request by assing**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by sanjaiganesh** [comment]
Looking to sync up with Ashutosh to understand the resource model

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 1]
```diff
@@ -0,0 +1,2157 @@
...
+      },
+      "required": [
+        "location"
+      ]
+    },
+    "alert": {
+      "description": "An alert created in alert management service.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```
Comment (sanjaiganesh) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: >Resource [](start = 33, length = 8)

is this tracked or proxy ? can you update these accordingly..  (#/definitions/TrackedResource)

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 2]
```diff
@@ -0,0 +1,2157 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "GetById": {
+            "$ref": "./examples/Alerts_GetById.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.AlertsManagement/alerts/{alertId}/changestate": {
```
Comment (sanjaiganesh) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: >changestate [](start = 68, length = 11)

Couldn't we use PUT itself to update the alertState property ?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 3]
```diff
@@ -0,0 +1,2157 @@
...
+        },
+        "x-ms-examples": {
+          "Get": {
+            "$ref": "./examples/SmartGroups_GetById.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/smartGroups/{smartGroupId}/changeState": {
+      "post": {
```
Comment (sanjaiganesh) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: >post [](start = 7, length = 4)

Couldn't we use PUT itself to update the smartGroupState property ?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 1 above)
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: This is proxy resource. I have updated the swagger.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 4]
```diff
@@ -0,0 +1,2162 @@
...
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "scope": {
+      "name": "scope",
+      "description": "scope here is reourceId for which alert is created.",
```
Comment (nschonni) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: reourceId -> resourceId

Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Done

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 2 above)
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: As discussed offline.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 5]
```diff
@@ -0,0 +1,2162 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "GetById": {
+            "$ref": "./examples/Alerts_GetById.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.AlertsManagement/alerts/{alertId}/changestate": {
```
Comment (sanjaiganesh) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: >alerts/{alertId} [](start = 51, length = 16)

I have double checked. We can create subscription level resources using PUT.. for ex: we have subscription level deployments, role assignments etc.. 

Also, how are we going to create alerts in the first place ? 

Please add PUT for create and Patch for updating the state

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 3 above)
Comment (sanjaiganesh) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: same feedback. subscription level PUTs are allowed.. 
Please add PUT for create and Patch for updating the state
 


---
In reply to: [265821049](https://github.com/Azure/azure-rest-api-specs/pull/5384#discussion_r265821049) [](ancestors = 265821049)

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 5 above)
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: User can't create alerts. User creats alert rules using Microsoft.Insights API. Alert Rules create these alert instances.  We are consuming the alert intsances. Our API provides a way to fetch these alert instances. Alerts are proxy resources. While user creating the alert rules, they can specify the different resources. So there is no way to create a alert. User can only create alert rules.  There is no need for PUT API.

Our API is already GA. https://docs.microsoft.com/en-us/rest/api/monitor/alertsmanagement/alerts/changestate

In GA Version we are using POST for change state.
Why are you enforcing to use Patch for change state? As we discussed we don't want to change the behavior of the API for new version. 

We are just making alerts as extension resource for more granular access. Currently if user doesn't have access at subscription then they can't fetch alerts. Here the extension resource will be, the resource used while creating the alert rules. So if user has access to that particular resource still they can see alerts for that resource.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 3 above)
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: SmartGroups are also same as alerts. 

This is the link for RPSDK Interface for subscription wide resources. 
https://msazure.visualstudio.com/One/_git/AzureUX-ResourceProviderSDK?path=%2Fsrc%2FResourceProvider%2FRouting%2FRouting%2FResourceManagement%2FISubscriptionWideResourceTypeHandler.cs&version=GBdev

It contains only GET and POST. We can't change smart groups change state to PATCH. The API is already in public preview. If you want to discuss, we can schedule a meeting.

Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: If you want I can remove all other resources from this API version. This API version will be used only for alerts and alertsSummary.

Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: I have removed all other resources.

**Comment by dsgouda** [comment]
Rest spec to Net Sdk code generation is not supported for new RPs, they will be onboarded separately. Please do not block on the `Automation for azure-sdk-for-net` failure for this review.

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 6]
```diff
@@ -0,0 +1,1062 @@
+{
+  "swagger": "2.0",
```
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: For now I am changing only alert and alertsSummary. There was no change for smart groups. So I thought I should remove that. I will add smart groups.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 7]
```diff
@@ -0,0 +1,1062 @@
...
+      }
+    }
+  },
+  "parameters": {
+    "scope": {
+      "name": "scope",
+      "description": "scope here is resourceId for which alert is created.",
+      "type": "string",
+      "in": "path",
+      "required": true
```
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: I was not aware of this. I will add this.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json [hunk 8]
```diff
@@ -0,0 +1,1062 @@
...
+        "x-ms-examples": {
+          "Summary": {
+            "$ref": "./examples/Alerts_Summary.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "scope": {
```
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: No

[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Location is not required.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/examples/Alerts_ChangeState.json [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  "parameters": {
+    "api-version": "2019-03-01-preview",
+    "scope": "subscriptions/3b540246-808d-4331-99aa-917b808a9166/resourcegroups/servicedeskresourcegroup/providers/microsoft.insights/components/servicedeskappinsight",
+    "alertId": "66114d64-d9d9-478b-95c9-b789d6502100",
+    "newState": "Acknowledged",
+    "body": {
+      "comments": "Acknowledging alert"
+    }
+  },
+	"responses": {
```
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/examples/Alerts_ChangeState.json]: Done

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Done

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Done

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (assing) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Done

**Comment by praries880** [comment]
@antmarti-microsoft Can you validate that the latest commit took care of your feedback?

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json (hunk 5 above)
Comment (sanjaiganesh) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json]: Sounds good. makes sense to me. I did miss on the part of that user can't create alerts.

**Review by anthony-c-martin — APPROVED** [review]

**Comment by assing** [comment]
@praries880 I think I can't merge this PR. Can you please merge this PR?

**Review by praries880 — APPROVED** [review]

**Pull request merged**